File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/SvgToPdf.cs
using System;
using System.IO;
using Xfinium.Pdf;
using Xfinium.Pdf.Graphics;
using Xfinium.Pdf.Graphics.Svg;

namespace Xfinium.Pdf.Samples
{
    /// <summary>
    /// Svg to Pdf conversion sample.
    /// </summary>
    public class SvgToPdf
    {
        /// <summary>
        /// Main method for running the sample.
        /// </summary>
        public static SampleOutputInfo[] Run(Stream input)
        {
            PdfFixedDocument document = new PdfFixedDocument();
            PdfPage page = document.Pages.Add();

            PdfSvgDrawing svg = new PdfSvgDrawing(input);
            page.Graphics.DrawFormXObject(svg, 20, 20, page.Width - 40, page.Width - 40);

            SampleOutputInfo[] output = new SampleOutputInfo[] { new SampleOutputInfo(document, "xfinium.pdf.sample.svgtopdf.pdf") };
            return output;
        }
    }
}

File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/SvgToPdf.vb
Imports System.IO
Imports Xfinium.Pdf
Imports Xfinium.Pdf.Graphics
Imports Xfinium.Pdf.Graphics.Svg

Namespace Xfinium.Pdf.Samples
	''' <summary>
	''' Svg to Pdf conversion sample.
	''' </summary>
	Public Class SvgToPdf
		''' <summary>
		''' Main method for running the sample.
		''' </summary>
		Public Shared Function Run(input As Stream) As SampleOutputInfo()
			Dim document As New PdfFixedDocument()
			Dim page As PdfPage = document.Pages.Add()

			Dim svg As New PdfSvgDrawing(input)
			page.Graphics.DrawFormXObject(svg, 20, 20, page.Width - 40, page.Height - 40)

			Dim output As SampleOutputInfo() = New SampleOutputInfo() {New SampleOutputInfo(document, "xfinium.pdf.sample.svgtopdf.pdf")}
			Return output
		End Function
	End Class
End Namespace
